### PR TITLE
Add data transfer and component object

### PR DESCRIPTION
**Pull request by Slim1011**
Add data transfer from json file and component object model

### CODE REVIEW

**Review by AhnessP — APPROVED** [review]

**Review by SerhiiTarasiuk — COMMENTED** [review]
Maybe we need to create way to organize or group our tests. 
And I wonder if it is normal to keep all test data in one file.

**Review by SerhiiTarasiuk — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -1,445 +0,0 @@
-# User-specific files
```
Comment (dmatsev) [inline .gitignore:1]: why did you remove the .gitignore file?

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,4 @@
+﻿[*.cs]
+
+# Default severity for analyzer diagnostics with category 'Style'
+dotnet_analyzer_diagnostic.category-Style.severity = silent
```
Comment (dmatsev) [inline .editorconfig]: why severity = silent?

**Review by dmatsev — CHANGES_REQUESTED** [review]
@Slim1011 Please revert .gitignore and Readme.md files.

File: EasyRestProjectNetTeam2/EasyRestComponentsObj/HeaderMenuComponent.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            driver.Navigate().Refresh();
+            return this;
+        }
+        public HeaderMenuComponent ClickSignInButton()
+        {
+            _signInButton.Click();
+            return this;
+        }
+
+        public IWebElement GetProfileIcon()
```
Comment (dmatsev) [inline EasyRestProjectNetTeam2/EasyRestComponentsObj/HeaderMenuComponent.cs]: You can't return IWebElement from Page Object/Component because it breaks the Incapsulation principle. Please fix this everywhere where you did it.

File: EasyRestProjectNetTeam2/EasyRestPages/BasePage.cs [hunk 4]
```diff
@@ -9,14 +10,13 @@ namespace EasyRestProjectNetTeam2.EasyRestPages
 {
     public class BasePage
     {
-        readonly IWebDriver driver;
+        readonly protected IWebDriver driver;
```
Comment (dmatsev) [inline EasyRestProjectNetTeam2/EasyRestPages/BasePage.cs:12]: why did you add `protected`?

File: EasyRestProjectNetTeam2/EasyRestPages/HomePage.cs [hunk 5]
```diff
@@ -1,20 +1,22 @@
-﻿using OpenQA.Selenium;
+﻿using EasyRestProjectNetTeam2.EasyRestComponentsObj;
+using OpenQA.Selenium;
 using SeleniumExtras.PageObjects;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
+using System.Threading;
```
Comment (dmatsev) [inline EasyRestProjectNetTeam2/EasyRestPages/HomePage.cs]: why do you need this?

File: EasyRestProjectNetTeam2/EasyRestPages/BasePage.cs [hunk 6]
```diff
@@ -35,11 +35,21 @@ public void WaitElementIsEnable(int timeToWait, IWebElement element)
             wait.Until(driver => element.Enabled);
         }
 
+        public void WaitElementIsClickable(int timeToWait, IWebElement element)
```
Comment (dmatsev) [inline EasyRestProjectNetTeam2/EasyRestPages/BasePage.cs]: That method could NOT be public because it has the `IWebElement` object as a parameter. Please consider the correct access modifier of that method or just do not pass IWebElement as a parameter.
@Slim1011

File: EasyRestProjectNetTeam2/Helpers/DataReader.cs [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+using System.Threading.Tasks;
+using EasyRestProjectNetTeam2.Models;
+
+namespace EasyRestProjectNetTeam2.Helpers
+{
+    internal class DataReader
+    {
+        private string _pathToJsonFile = Directory.GetParent(@"../../../").FullName + "\\TestsData\\TestsData.json";
+
+        public void ReadData(out DataModel dataModelObject)
```
Comment (dmatsev) [inline EasyRestProjectNetTeam2/Helpers/DataReader.cs]: why do you use `out` construction?

**Review by eugenek7 — APPROVED** [review]

**Review by katemalash — APPROVED** [review]

File: EasyRestProjectNetTeam2/Reports/index.html [hunk 8]
```diff
@@ -49,7 +49,7 @@
           </li>
             <li class="m-r-10">
                 <a href="#">
-                    <span class="badge badge-primary">08.07.2022 18:07:52</span>
+                    <span class="badge badge-primary">13.07.2022 13:50:57</span>
```
Comment (dmatsev) [inline EasyRestProjectNetTeam2/Reports/index.html]: please remove html files from the repository and add them to the .gitignore

**Review by dmatsev — CHANGES_REQUESTED** [review]
Please resolve the comments.

**Review by dmatsev — APPROVED** [review]
Awesome!
